rdt/transformers/categorical.py

```diff
@@ -195,15 +194,15 @@ def _transform(self, data):
     def _reverse_transform_by_matrix(self, data):
         """Reverse transform the data with matrix operations."""
         num_rows = len(data)
-        num_categories = len(self.means)
+        num_categories = len(self.starts)
 
         data = np.broadcast_to(data, (num_categories, num_rows)).T
-        means = np.broadcast_to(self.means, (num_rows, num_categories))
-        diffs = np.abs(data - means)
-        indexes = np.argmin(diffs, axis=1)
+        starts = np.broadcast_to(self.starts.index, (num_rows, num_categories))
+        diffs = (data >= starts)[:, ::-1]
+        indexes = num_categories - np.argmax(diffs, axis=1) - 1
```

Comment:
What exactly are these two like doing?

Reply:
cool, can we rename diffs to `interval_starts_data_is_greater_than` or something like that? and `indexes` to `interval_indexes`?